src/chunk_scan.c

```diff
@@ -313,15 +313,25 @@ ts_chunk_scan_by_constraints(const Hyperspace *hs, const List *dimension_vecs,
 			chunk_count++;
 		}
 	}
+
 	/* get the osm chunk if needed */
 	if (osm_chunk_id != INVALID_CHUNK_ID && !found_osm_chunk)
 	{
 		MemoryContext old_mcxt;
 		old_mcxt = MemoryContextSwitchTo(orig_mcxt);
+		/* Lazy initialize the chunks array */
+		if (NULL == chunks)
+		{
+			int chunks_alloc = unlocked_chunk_count;
+			if (osm_chunk_id != INVALID_CHUNK_ID && !found_osm_chunk)
+				chunks_alloc = unlocked_chunk_count + 1;
+			chunks = MemoryContextAlloc(orig_mcxt, sizeof(Chunk *) * chunks_alloc);
```

Comment:
There are no tests for this. Do you need to add some?

Reply:
@mkindahl this is OSM code path. I am just merely fixing the clang warning here. I have no idea how to add a test for this.

Reply:
I wonder if this is a false positive and `chunks` can ever be null.

Reply:
I actually think this was a bug and this is a fix.

The `chunks` variable can be null if `unlocked_chunk_count` is zero or if the loop preceding this check failed to lock all unlocked chunks. I think you can move the lazy init to before the loop above, which avoids the duplicate code, makes the code easier to maintain, and also get rid of the bug.

Reply:
@mkindahl this code path has been changed by Akuzm's latest commit. So no more warnings in this code path.

Reply:
Ok.